specs/ceph_pool_usage_data.adoc

```diff
@@ -0,0 +1,175 @@
+and store the same in central store.
+
+Additionally trending graphs as well are required for pool utilization data to
+help administrators to decide on expansion of the cluster.
+
+
+== Problem description
+
+Currently the ceph integration module synchronizes the cluster data to its
+central store. The cluster data should also contain the pool utilization data
+fetched and maintained with cluster so that same can be projected in UI for more
+help to administrator.
+
+In addition to instant pool utilization data maintained with cluster data, the
+time series data also should be maintained in Tendrl for rendering the trending
+graphs for pool utilizations.
+
+
+== Use Cases
+
+* This specification covers the Trending graphs in Tendrl for ceph pool
+utilization data
+
+* The cluster details should contain pool utilization data as well
+
+
+== Proposed change
+
+* Introduce a collectd plug-in which collects the pools utilization data from
+the underlying cluster and pushes the same graphite time series DB
+
+* Enhance the ceph cluster sync logic in Tendrl's ceph integration module to
+fetch process and update the pool utilization data to central store with
+cluster's pool inventory.
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+* Update the `Pool` data model to include utilization related fields `used`,
+`total` and `PercentUsed`
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+* The pool listing API `GET /cluster/:cluster_id/pool/:pool_id` should list the
+pool with additional fields related to utilization as below
+
+```
+Status: 200 OK
+
+{
+    'cluster_id': 'bc2c51e1-516d-45b8-8310-7caab2fa784c',
+    'min_size': '1',
+    'pg_num': '128',
+    'pool_id': '1',
+    'pool_name': 'testpool',
+    'utilization': {
+        'used': 0,
+        'total': 1024,
+        'percent_used': 0
+    }
+}
+```
+
+==== Notifications/Monitoring impact:
+
+* Add a collectd plug-in in monitoring module for getting the pool utilization
+data and pushing it to time series DB
+
+==== Tendrl/common impact:
+
+None
+
+==== Tendrl/node_agent impact:
+
+None
+
+==== Sds integration impact:
+
+* Tendrl's ceph integration module should synchronize the pool utilization data
+as well while cluster data getting synchronized
+
+=== Security impact:
+
+None
+
+=== Other end user impact:
+
+* The REST output for listing individual pools would be including additional
+utilization details
+
+=== Performance impact:
+
+* As confirmed by ceph team, the execution of `ceph df` command on leader mon is
+not resource intensive.
+
+* The collectd plug-in frequency should not be set to anything less than 5
+seconds as ceph cluser itself syncs data at this interval
+
+=== Other deployer impact:
+
+None
+
+=== Developer impact:
+
+* API module owner to make sure pool listing contains the utilization details
+as well and shown properly
+
+== Implementation:
+
+* Add a collectd plug-in to execute `ceph df` command to fetch the pool
```

Comment:
This brings up the problem of localising collectd plugins. The monitoring provisioning flows would not currently allow node specific flows. They'll be possible once node agent tagging and tag based job routing are implemented via https://github.com/Tendrl/specifications/issues/44. This is another reason for restricting this particular specification to just the bridge implementation since that isn't blocked.